code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -191,8 +266,28 @@
 	A.launch(target, def_zone)
 
 /mob/living/simple_animal/hostile/proc/DestroySurroundings()
+	if(istype(src, /mob/living/simple_animal/hostile/megafauna))
+		dir = get_dir(src,target_mob)
```

Comment:
`set_dir(get_dir(src,target_mob))`
stuff depends on mobs only changing dirs and locs via setters like that, can possibly break things if bypassed.